UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareServiceCache.cs

```diff
@@ -66,19 +67,33 @@ public async Task<List<Products>> GetNonCachedProductDataForFss(List<Products> p
                     UpdateNumbers = new List<int?>(),
                     Bundle = item.Bundle
                 };
+
                 List<int?> updateNumbers = new List<int?>();
                 foreach (var itemUpdateNumber in item.UpdateNumbers)
                 {
                     var compareProducts = $"{item.ProductName}|{item.EditionNumber.Value}|{itemUpdateNumber.Value}";
                     var productList = new List<string>();
+
                     if (!productList.Contains(compareProducts))
                     {
                         var storageConnectionString = azureStorageService.GetStorageAccountConnectionString(fssCacheConfiguration.Value.CacheStorageAccountName, fssCacheConfiguration.Value.CacheStorageAccountKey);
                         var cacheInfo = (FssSearchResponseCache)await azureTableStorageClient.RetrieveFromTableStorageAsync<FssSearchResponseCache>(item.ProductName, item.EditionNumber + "|" + itemUpdateNumber.Value, fssCacheConfiguration.Value.FssSearchCacheTableName, storageConnectionString);
                         if (cacheInfo != null && !string.IsNullOrEmpty(cacheInfo.Response))
                         {
                             var internalBatchDetail = await CheckIfCacheProductsExistsInBlob(exchangeSetRootPath, queueMessage, item, updateNumbers, itemUpdateNumber, storageConnectionString, cacheInfo);
-                            internalSearchBatchResponse.Entries.Add(internalBatchDetail);
+
+                            updateNumbers.Clear();
```

Comment:
updateNumbers.Clear() has been moved and is now cleared before it's checked. Could you confirm if that's correct please. Would it make more sense before the call to CheckIfCacheProductsExistsInBlob?

Reply:
That was wrong, Corrected it.

Reply:
Thanks for correcting, but I think the following two tests are now failing because of the correction, could you check them:

 - WhenGetNonCachedProductDataForFssIsCalled_ThenReturnProductNotFound
 - WhenGetNonCachedProductDataForFssIsCalled_ThenReturnProductNotFoundForLargeMediaExchangeSet

They both expect 1, but are actually getting 0

Reply:
Resolved.